elide-core/src/main/java/com/yahoo/elide/core/PersistentResource.java

```diff
@@ -888,7 +892,10 @@ protected boolean checkRelation(String relationName) {
      */
     private Set<PersistentResource> getRelationUnchecked(String relationName, Set<Predicate> filters) {
         RelationshipType type = getRelationshipType(relationName);
-        Object val = getValueUnchecked(relationName);
+        final Class<?> entityClass = dictionary.getParameterizedType(obj, relationName);
```

Comment:
From @xiaoyao1991: http://stackoverflow.com/questions/9921676/what-does-class-mean-in-java